lib/ability.rb

```diff
@@ -29,6 +29,7 @@ def initialize(user, resource, controller = nil)
       end
 
       cannot(:switch_to, User) { |target_user| !target_user.active? }
+      ability_extender.extend(user, resource)
```

Comment:
This doesn't strictly need to be here to make this feature work, but I feel like it should have it. The fact that ability extensions weren't applied in all scenarios confused me until I remembered the early `return` for global admins.